### PR TITLE
[Draft] DIV-8511 | new signer for health professional use case

### PR DESCRIPTION
**Pull request by varadeth**
@samarthgr @egov-joy @prasanna-egov Please review

### CODE REVIEW

File: backend/hpr_certificate_signer/config/Template.template [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (egov-joy) [inline backend/hpr_certificate_signer/config/Template.template:1]: Shouldn't this be in etcd?

File: backend/hpr_certificate_signer/config/constant.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const BAD_REQUEST = 400;
```
Comment (egov-joy) [inline backend/hpr_certificate_signer/config/constant.js:1]: filename should be constants.js

Comment (egov-joy) [inline backend/hpr_certificate_signer/config/constant.js:1]: filename should be constants.js

File: backend/hpr_certificate_signer/config/config.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+const KAFKA_BOOTSTRAP_SERVER = process.env.KAFKA_BOOTSTRAP_SERVERS || 'localhost:9092';
+const KAFKA_CONSUMER_SESSION_TIMEOUT = process.env.KAFKA_CONSUMER_SESSION_TIMEOUT || 300000; // in ms
+const HEALTH_CERTIFY_TOPIC = process.env.HEALTH_CERTIFY_TOPIC || 'health_certify';
```
Comment (egov-joy) [inline backend/hpr_certificate_signer/config/config.js:3]: Let's reuse certify here as discussed.

Comment (egov-joy) [inline backend/hpr_certificate_signer/config/config.js:3]: Let's reuse certify here as discussed.

File: backend/Makefile [hunk 4]
```diff
@@ -12,5 +12,6 @@ docker:
 	$(MAKE) -C certificate_signer
 	$(MAKE) -C test_certificate_signer
 	$(MAKE) -C certificate_api
+	$(MAKE) -C hpr_certificate_signer
```
Comment (egov-joy) [inline backend/Makefile:15]: Let's rename this to generic_certificate_signer as this is a step towards making a generic signer.

Comment (egov-joy) [inline backend/Makefile:15]: Let's rename this to generic_certificate_signer as this is a step towards making a generic signer.

File: docker-compose.yml [hunk 5]
```diff
@@ -29,7 +29,7 @@ services:
     volumes:
       - ./flagr_dump.sql:/docker-entrypoint-initdb.d/flagr_dump.sql
   registry:
-    image: divoc/registry-es
+    image: divoc/registry
```
Comment (egov-joy) [inline docker-compose.yml:32]: Why is this change required?

Comment (egov-joy) [inline docker-compose.yml:32]: Why is this change required?

File: backend/hpr_certificate_signer/config/Template.template (hunk 1 above)
Comment (varadeth) [reply backend/hpr_certificate_signer/config/Template.template:1]: yes. it should be in. it was only for testing purpose. I will change this to get this fetched from etcd

File: docker-compose.yml (hunk 5 above)
Comment (varadeth) [reply docker-compose.yml:32]: no. will remove this

File: backend/generic_certificate_signer/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,14 @@
+FROM node:lts-alpine
+WORKDIR /app
+ENV PATH /app/node_modules/.bin:$PATH
+ENV CERTIFICATE_NAMESPACE "https://gist.githubusercontent.com/varadeth/fff3a5a48bdf0adb4845410d507e97da/raw/8e5fb1a92c66f6fa0cd9a5d8cc8bed714ef97273/context.js"
+ENV CERTIFICATE_CONTROLLER_ID "https://cowin.gov.in/"
+ENV CERTIFICATE_PUBKEY_ID "https://example.com/i/india"
+ENV CERTIFICATE_DID "did:india"
```
Comment (prasanna-egov) [inline backend/generic_certificate_signer/Dockerfile:7]: lets remove india reference here.

File: backend/generic_certificate_signer/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,14 @@
+FROM node:lts-alpine
+WORKDIR /app
+ENV PATH /app/node_modules/.bin:$PATH
+ENV CERTIFICATE_NAMESPACE "https://gist.githubusercontent.com/varadeth/fff3a5a48bdf0adb4845410d507e97da/raw/8e5fb1a92c66f6fa0cd9a5d8cc8bed714ef97273/context.js"
```
Comment (prasanna-egov) [inline backend/generic_certificate_signer/Dockerfile:4]: Reference need to change to either egov repo

File: backend/generic_certificate_signer/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+ENV CERTIFICATE_CONTROLLER_ID "https://cowin.gov.in/"
+ENV CERTIFICATE_PUBKEY_ID "https://example.com/i/india"
+ENV CERTIFICATE_DID "did:india"
+ENV CERTIFICATE_ISSUER "https://cowin.gov.in/"
```
Comment (prasanna-egov) [inline backend/generic_certificate_signer/Dockerfile:8]: again this can be a generic URL like divoc.xiv.in or something like that

File: backend/generic_certificate_signer/config/config.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+const CERTIFICATE_CONTROLLER_ID = process.env.CERTIFICATE_CONTROLLER_ID || 'https://divoc.dev/';
+const CERTIFICATE_ISSUER = process.env.CERTIFICATE_ISSUER || "https://divoc.dev/";
+const CERTIFIED_TOPIC = process.env.CERTIFIED_TOPIC || 'health_certified';
+const DUPLICATE_CERTIFICATE_TOPIC = process.env.DUPLICATE_CERTIFICATE_TOPIC || 'duplicate_certificate';
+const CERTIFICATE_ACK_TOPIC = process.env.CERTIFICATE_ACK_TOPIC || 'health_certify_ack';
+const REGISTRY_CERTIFICATE_SCHEMA = process.env.REGISTRY_CERTIFICATE_SCHEMA || 'HealthProfessional';
+const CERTIFICATE_RETRY_COUNT = process.env.CERTIFICATE_RETRY_COUNT || 5;
+const REGISTRY_URL = process.env.REGISTRY_URL || 'http://localhost:8081/';
+const REDIS_ENABLED = process.env.REDIS_ENABLED || true;
+const REDIS_URL = process.env.REDIS_URL || 'redis://0.0.0.0:6379';
```
Comment (prasanna-egov) [inline backend/generic_certificate_signer/config/config.js:15]: should be localhost not 0.0.0.0

File: backend/generic_certificate_signer/config/config.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+const REDIS_KEY_EXPIRE = process.env.REDIS_KEY_EXPIRE || 2 * 24 * 60 * 60; // in secs
+const CERTIFICATE_NAMESPACE = process.env.CERTIFICATE_NAMESPACE || 'https://gist.githubusercontent.com/varadeth/fff3a5a48bdf0adb4845410d507e97da/raw/3799da91f3968ec4eaf1783775ee51cc60098e61/context.json';
```
Comment (prasanna-egov) [inline backend/generic_certificate_signer/config/config.js:17]: file should be from generic repo not Tejas' personal repo

File: default-configuration/etcd/HP_TEMPLATE.template [hunk 11]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (egov-joy) [inline default-configuration/etcd/HP_TEMPLATE.template:1]: refer to #1112 for a new structure of managing entity type specific configurations. We can reuse that structure for this template as well.
